docs/applications/add-application-deploy-status-to-commit.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Add Application Deploy Status to Commit
+intro: Commit statuses provide a way to tag commits with meta data, like automated build results. Learn how to create outgoing webhooks to update a commit status to match your Devopness application deploy status.
+links:
+    overview:
+    quickstart:
+    previous:
+    next:
+    guides:
+    related:
+    featured:
+---
+
+NOTE: Webhooks, for now, are an API only feature; so this post will guide you through the usage of our API to help you create an outgoing webhook.
+
+NOTE: The examples below use `cURL`, which is a command line utility to be used from a terminal window. If you're not familiar with running commands on a terminal or you simply don't want to use a terminal you could, alternatively, copy the examples and import the `cURL` command line into your favorite API client platform. Here are links with instructions on how to achieve that on some of the most popular API client platforms:
+
+- Postman: [Importing and exporting data: Importing with cURL commands](https://learning.postman.com/docs/getting-started/importing-and-exporting-data/#importing-with-curl-commands)
+- Insomnia: [Import and Export Data: Import Data](https://docs.insomnia.rest/insomnia/import-export-data#import-data)
+
+NOTE: Insomnia uses curly braces syntax for environment variables, to avoid errors you need to disable the nunjuncks template feature for the request body. Further instructions at [Insomnia's FAQ](https://docs.insomnia.rest/insomnia/faq#how-can-i-temporarily-disable-nunjucks-template)
```

Comment:
Post updated
![image](https://github.com/devopness/devopness/assets/33532177/34490dd6-3900-4dac-aab5-2f538e2b0d17)

Reply:
@thlmenezes I reviewed again and no changes related to this conversation. Are you sure you updated the branch?

Reply:
Sorry @jfoliveira I misinterpret this thread the first time I've read.

I disagree

The note about insomnia is needed to help the user succeed.

The following UX follows the `cURL` request import on insomnia

![error_insomnia](https://github.com/devopness/devopness/assets/33532177/901d96f1-d17a-4e60-9ba4-17aa92f5388c)

And if you try to send the request

![error_insomnia_req](https://github.com/devopness/devopness/assets/33532177/87126891-4bad-4639-9788-4eea32a45bcc)

Many users may use insomnia, since it's a popular tool, and be lost when encountering this error, because the error message and its workaround solution are not obvious; that was the reasoning behind the documentation, to help the user succeed in making the request on all the options cited on the post.

> That's expensive to maintain

I agree, but in this case I believe is needed, since the first moment after importing the data the user is met with an error.

If we remove this note, I suggest removing all Insomnia references

Does this seem like reasonable justification to maintain this note in docs, @jfoliveira ?

Reply:
> The note about insomnia is needed to help the user succeed.

The third-party tool own documentation is not enough to help users, who decided to use that tool, to achieve success on their tool?

> Many users may use insomnia
True. Many MAY, many other MAY NOT. Should our docs care about all possible tools that our users MAY use?
Should we be monitoring market share and every time another tool become popular we need to update our docs?
Can you see that as a sustainable approach?

>     That's expensive to maintain
>> I agree, but in this case I believe is needed,

I'm not convinced, so I left some food for thought for you above and also:

1. Should Devopness team pay the price of documenting edge cases on third-party tools being used by specific users, by their own choice? Are you sure shouldn't the user read the documentation of the tools they decided to use?
2. If the user is not able to read the docs of the external tool they chose, why would they bother reading Devonpess documentation?
3. Likewise, are you also expecting third-party tools to also document Devopness specific behaviors every time we change things on our side?

Please don't take too much time replying to every single topic. If you're convinced, please just summarize the reason why you believe the approach you are suggesting is a common practice in the software industry.

Reply:
@thlmenezes one extra point on your note:

```
**NOTE**: Insomnia uses curly braces syntax for environment variables, to avoid errors you need to 
disable the nunjuncks template feature for the request body. Further instructions at [Insomnia's
```

is bringing a lot of new words and concepts (`curly braces`, `syntax`, `environment variables`, `nunjunkcs`, `template`) to a user who just arrived at this topic with a single goal: create a Devopness outgoing webhook.

Does the average user (which doesn't use or need such specific third-party tool) need to be presented to all of this info?

Reply:
@jfoliveira removed note from docs.

May the users discover and solve this with the Insomnia team.